composites/SnippetPreview/components/Button.js

```diff
@@ -62,3 +68,50 @@ export const SnippetPreviewButton = styled( BaseButton )`
 	line-height: 15px;
 	font-size: 0.8rem;
 `;
+
+/**
+ * Applies styles to icon for IconButton with text.
+ *
+ * @param {ReactElement} icon The original icon.
+ *
+ * @returns {ReactElement} Icon with text styles applied.
+ */
+function applyIconTextStyle( icon ) {
+	return styled( icon )`
+		margin: 0 8px 0 0;
+		float: left;
+	`;
+}
+
+/**
+ * Returns an icon button that can optionally contain text.
+ *
+ * @param {object} props Component props.
+ *
+ * @returns {ReactElement} Styled icon button.
+ */
+export const IconButton = ( props ) => {
+	const { children, icon, iconColor } = props;
+
+	let IconComponent = Icon;
```

Comment:
Because the Icon is an import I cannot assign a new value to it, and since I optionally have to add styles to the Icon in case there is text I need to put it in a new variable.